vespa-osgi-testrunner/src/main/java/com/yahoo/vespa/testrunner/VespaCliTestRunner.java

```diff
@@ -126,14 +132,35 @@ ProcessBuilder testRunProcessBuilder(Suite suite, TestConfig config) throws IOEx
         builder.environment().put("VESPA_CLI_HOME", ensureDirectoryForVespaCli("cli-home").toString());
         builder.environment().put("VESPA_CLI_CACHE_DIR", ensureDirectoryForVespaCli("cli-cache").toString());
         builder.environment().put("VESPA_CLI_ENDPOINTS", toEndpointsConfig(config));
-        Path certRoot = certificateRoot(config);
-        builder.environment().put("VESPA_CLI_DATA_PLANE_KEY_FILE", certRoot.resolve("key").toAbsolutePath().toString());
-        builder.environment().put("VESPA_CLI_DATA_PLANE_CERT_FILE", certRoot.resolve("cert").toAbsolutePath().toString());
+        Credentials credentials = getCredentials(config);
+        builder.environment().put("VESPA_CLI_DATA_PLANE_KEY_FILE", credentials.privateKeyFile().toString());
+        builder.environment().put("VESPA_CLI_DATA_PLANE_CERT_FILE", credentials.certificateFile().toString());
         return builder;
     }
 
-    private Path certificateRoot(TestConfig config) {
-        return config.system().isPublic() ? artifactsPath : Path.of(Defaults.getDefaults().underVespaHome("var/vespa/sia"));
+    private record Credentials(Path privateKeyFile, Path certificateFile) {}
+
+    private Credentials getCredentials(TestConfig config) {
+        final Path privateKeyFile;
+        final Path certificateFile;
+        if (config.system().isPublic()) {
+            privateKeyFile = artifactsPath.resolve("key");
+            certificateFile = artifactsPath.resolve("cert");
+        } else {
+            Path siaRoot = vespaHome.resolve("var/vespa/sia");
+            List<AthenzIdentity> services = SiaUtils.findSiaServices(siaRoot);
+            if (services.isEmpty()) {
+                throw new IllegalArgumentException("No service credentials in " + siaRoot + ". Application has no " +
```

Comment:
Will exceptions thrown here be visible for end-user, or should failure be added to log/test report?

Reply:
This is called inside `runTests`, which catches `Exception` and creates a log report for it.